stacks-signer/src/runloop.rs

```diff
@@ -136,24 +141,80 @@ pub struct RunLoop<C> {
+            warn!("Signer index not found. Have you registered your signing key yet?");
+            return Err(ClientError::NotRegistered);
+        };
+        info!("Signer index found. Initializing signer {index}...");
+        self.signing_round.signer_id = index as u32;
+        // TODO: REMOVE THIS AWFULNESS! HIDE THE SHAME! DO NOT LOOK AT ME....
+        // Just a temporary placeholder until we have proper config retrieval from the node.
+        // Then for the love of all that is holy, rip this out and burn it with fire.
+        let signers = self.stacks_client.get_signers().unwrap_or(vec![]);
+        let mut new_signers = HashMap::new();
+        let old_addresses = self
+            .signing_round
+            .public_keys
+            .signers
+            .iter()
+            .map(|(old_id, old_public_key)| {
+                let bytes = old_public_key.to_bytes();
+                let signer_hash = Hash160::from_data(&bytes);
+                let signing_address = StacksAddress::p2pkh_from_hash(self.mainnet, signer_hash);
+                (old_id, signing_address)
+            })
+            .collect::<Vec<_>>();
+        for (new_id, (stacks_address, _num_slots)) in signers.iter().enumerate() {
+            if let Some((old_id, old_address)) = old_addresses
+                .iter()
+                .find(|(_old_id, address)| address == stacks_address)
+            {
+                if stacks_address == old_address {
+                    new_signers.insert(
+                        new_id as u32,
+                        self.signing_round
+                            .public_keys
+                            .signers
+                            .get(*old_id)
+                            .expect("BUG: this should never happen")
+                            .clone(),
+                    );
+                }
+            }
+        }
+        self.signing_round.public_keys.signers = new_signers;
+        if self.signing_round.public_keys.signers.is_empty() {
+            info!("Failed to retrieve list of signers from stacker db. Have you registered your signing key yet?");
+            return Err(ClientError::NotRegistered);
+        }
+        self.update_dkg()?;
+        Ok(())
+    }
+
+    /// Retrieve the aggregate public key from the stacks node and trigger DKG if required
+    fn update_dkg(&mut self) -> Result<(), ClientError> {
+        let vote_state = self.stacks_client.get_current_vote_state()?;
+        if let Some(key) = vote_state.aggregate_public_key {
+            debug!("Updating aggregate public key: {:?}", key);
             self.coordinator.set_aggregate_public_key(Some(key));
-        } else {
-            debug!("Aggregate public key is not set. Coordinator must trigger DKG...");
-            // Update the state to IDLE so we don't needlessy requeue the DKG command.
+        } else if self.state == State::Uninitialized
+            || vote_state.total_votes >= self.dkg_threshold as u128
```

Comment:
If `vote_state.total_votes >= dkg_threshold` doesn't that mean that the vote succeeded?

Reply:
No it could be that it failed. However, this logic is still wrong. It should be checking if how many votes have been distributed among multiple options. so I will remove this check....need some other means to know "vote was started but consensus not reached vs vote never started"